pkg/importer/avro/tests/complex_test.arrai

```diff
@@ -0,0 +1,135 @@
+        tagList <: sequence of string[default="['tag1', 'tag2']"]
+        len <: md5
+        width <: width
+        occurrenceDateTime <: datetime[~timestamp-millis, description="The date and time, to millisecond precision, that this event occurred"]
+        amount <: decimal(4.2)[description="Amount", default="0"]
+
+
+    !type Header:
+        eventUUID <: EventUUID
+
+
+    !type Owner:
+        id <: string
+        ownerTags <: sequence of OwnerTag?
+
+
+    !type OwnerTag:
+        id <: string
+        tag <: string
+
+
+    !type Party[description="Party Details"]:
+        id <: string[description="A unique identifier for party"]
+        subject <: Subject
+
+
+    !type Relationship[namespace="com.example", description="account relationship."]:
+        owner <: Owner?
+        performedAction <: Action
+        relationshipType <: string[description="This would be the business key for uniquely identifying a relation"]
+        startDate <: string?[description="Relationship start date", default="null"]
+        endDate <: string?[union=['string', 'null'], description="Relationship end date", default="null"]
+        issueDate <: IntString?[union=['int', 'string', 'null'], description="Relationship issue date", default="null"]
+        closeDate <: StringInt?[description="Relationship close date", default="null"]
+        realCloseDate <: StringInt?[union=['string', 'null', 'int'], description="Relationship real close date", default="null"]
+        updateDate <: StringInt[description="Relationship update date", default="null", order="ascending"]
+        updateDate1 <: StringInt[description="Relationship update date", default="null", order="ascending", alias_of="updateDate"]
+        updateDate2 <: StringInt[description="Relationship update date", default="null", order="ascending", alias_of="updateDate"]
+        titleList <: sequence of string?[default="[]"]
+        tagList <: StringSequenceTagListItemSequence?[union=['StringSequence', 'null', 'tagListItemSequence'], default="[]"]
+        titleMap <: set of string
+        tagMap <: StringSetTagMapItemSet?
+
+    !alias P1:
+        Relationship
+
+    !alias P2:
+        Relationship
+    !type StringMapRecordItem[json_map_key="key"]:
+        key <: string
+        value <: MapRecord
+
+
+    !type MapRecord:
+        id <: string
+
+
+    !type StringtagMapItemItem[json_map_key="key"]:
+        key <: string
+        value <: tagMapItem
+
+
+    !type tagMapItem:
+        id <: string
+        name <: string
+
+
+    !type Subject:
+        id <: string
+
+
+    !type tagListItem:
+        id <: string
+        name <: string
+
+
+    # Aliases
+
+    !alias md5[~fixed, namespace="com.example"]: bytes(16)
+    
+    !alias width[~fixed, namespace="com.example"]: bytes(16)
+    !alias width1: width
+    !alias width2: width
+    # Unions
+
+    !union IntString:
+        int
+        string
+
+    !union StringInt:
+        string
+        int
```

Comment:
Nit: Are these not the same thing? We shouldn't have both. If they're generated, sort the inner types by name so they end up the same thing.

Reply:
Good question.

The order of the types within the Sysl union is being used to represent the order of the types within Avro union. The `union` attribute is only added if the order can't be derived by the type (and nullable value) alone. 

I'm tempted to go with this suggestion. It will likely add additional `union` attributes in the general case but I think it's a cleaner representation. @ericzhang6222 I'll update specification.

Reply:
@ericzhang6222 Specification updated.